### ISSUE TITLE
All Autocomplete components open on mount

### ISSUE THREAD

**Issue by sebastianvitterso** [equinor/design-system#2563]
**Describe the bug**

We have multiple `<Autocomplete>` component on the landing page of our application, and the application is loaded, these open up automatically (which isn't very logical to me). 

![image](https://user-images.githubusercontent.com/37065184/194842305-5afd553b-36d9-49e2-b036-12591066e52b.png)

**Steps to reproduce the bug**

1. Use an autocomplete
2. Have it open even when it hasn't been clicked

```jsx

<Autocomplete
  autoWidth
  options={users}
  label="User"
  onOptionsChange={(changes) => setUser(changes.selectedItems[0])}
/>
```

**Expected behavior**

All autocompletes should be closed until clicked.

**Specifications**

- Version: 0.20.4
- Browser: Electron 20.0.1 (Chromium 104.0.5112.81)
- OS: MacOS 12.6 (21G115)

**Additional context**

Add any other context or screenshots about the bug. An example in [CodeSandbox](https://codesandbox.io/) is very helpful

**Comment by mimarz** [comment]
Thanks for reaching out to us! 

This is unfortunately bug related to React 18 (useEffect being called double) and the [Downshift](https://github.com/downshift-js/downshift) library which we are using in the `Autocomplete`. Doesn't look like they have fixed it yet... However, this should only occurs in dev mode.

More information can be found here: #2369

**Comment by sebastianvitterso** [comment]
Any chance you could push a bit on this issue up-stream?

**Comment by mimarz** [comment]
> Any chance you could push a bit on this issue up-stream?

Done 👍

**Comment by AndreasPresthammer** [comment]
Had same problem. Only found this issue to occur inside a dialog.

Repro:
https://codesandbox.io/s/npm-playground-forked-zcby68

**Comment by oddvernes** [comment]
We'll take a look in case there is something other than the downshift/react18 issue going on in that particular scenario 👍

**Comment by oddvernes** [comment]
The main issue here should be resolved by #2697 
The dialog case is a separate issue due to the input recieving focus by the floating-ui `FloatingFocusManager` created a seperate ticket for this [here](https://github.com/equinor/design-system/issues/2698)
